makefiles/app_dirs.inc.mk

```diff
@@ -7,12 +7,19 @@ RIOTBASE ?= .
 # 2. use patsubst to drop trailing "/"
 # 3. use patsubst to drop possible leading "./"
 # 4. sort
-APPLICATION_DIRS := $(sort $(patsubst ./%,%,$(patsubst %/,%,$(dir $(wildcard \
-	$(RIOTBASE)/fuzzing/*/Makefile     \
-	$(RIOTBASE)/bootloaders/*/Makefile \
-	$(RIOTBASE)/examples/*/Makefile    \
-	$(RIOTBASE)/tests/*/Makefile       \
-	)))))
+
+# Prepare the list of application directories
+APPLICATION_DIRS :=                   \
+    fuzzing                           \
+    bootloaders                       \
+    examples                          \
+    tests                             \
+    tests/drivers                     \
+
+APPLICATION_DIRS := $(addprefix $(RIOTBASE)/,$(APPLICATION_DIRS))
+APPLICATION_DIRS_RELATIVE := $(dir $(wildcard $(addsuffix /*/Makefile,$(APPLICATION_DIRS))))
+APPLICATION_DIRS_ABSOLUTE := $(abspath $(APPLICATION_DIRS_RELATIVE))
+APPLICATION_DIRS := $(sort $(patsubst ./%,%,$(patsubst %/,%,$(APPLICATION_DIRS_RELATIVE))))
```

Comment:
@aabadie like this?

Reply:
yes!
After it will be easy to extend the list with new test subfolders (sys, pkg, periph, etc)